client/web-sveltekit/src/lib/repo/fileIcons.ts

```diff
@@ -0,0 +1,620 @@
+    KOTLIN = 'kt',
+    LOCKFILE = 'lock',
+    LUA = 'lua',
+    MARKDOWN = 'md',
+    MDX = 'mdx',
+    NCL = 'ncl',
+    NIX = 'nix',
+    NPM = 'npmrc',
+    OCAML = 'ml',
+    PHP = 'php',
+    PERL = 'pl',
+    PERL_PM = 'pm',
+    PNG = 'png',
+    POWERSHELL_PS1 = 'ps1',
+    POWERSHELL_PSM1 = 'psm1',
+    PYTHON = 'py',
+    R = 'r',
+    R_CAP = 'R',
+    RUBY = 'rb',
+    GEMFILE = 'Gemfile',
+    RUST = 'rs',
+    SCALA = 'scala',
+    SASS = 'scss',
+    SQL = 'sql',
+    SVELTE = 'svelte',
+    SVG = 'svg',
+    SWIFT = 'swift',
+    TEST = 'test',
+    TOML = 'toml',
+    TYPESCRIPT = 'ts',
+    TSX = 'tsx',
+    TEXT = 'txt',
+    YAML = 'yaml',
+    YML = 'yml',
+    ZIG = 'zig',
+}
+
+// We use DviClass to enumerate Devicon classes
+// These values will be interpolated in the actual component:
+//      <i class={`devicon-${DviClass.JAVA}-plain`}>
+// Visit https://devicon.dev/ for correct class names
+enum DviClass {
+    C = 'c',
+    CLOJURE = 'clojure',
+    CLOJURE_SCRIPT = 'clojurescript',
+    CPP = 'cplusplus',
+    CSHARP = 'csharp',
+    CSS = 'css3',
+    DART = 'dart',
+    DOCKER = 'docker',
+    FSHARP = 'fsharp',
+    GIT = 'git',
+    GO = 'go',
+    GRAPHQL = 'graphql',
+    GROOVY = 'groovy',
+    HASKELL = 'haskell',
+    HTML5 = 'html5',
+    JAVA = 'java',
+    JAVASCRIPT = 'javascript',
+    JSON = 'json',
+    JULIA = 'julia',
+    KOTLIN = 'kotlin',
+    LUA = 'lua',
+    NPM = 'npm',
+    OCAML = 'ocaml',
+    PERL = 'perl',
+    PHP = 'php',
+    PYTHON = 'python',
+    R = 'r',
+    REACT = 'react',
+    RUBY = 'ruby',
+    RUST = 'rust',
+    SASS = 'sass',
+    SCALA = 'scala',
+    SVELTE = 'svelte',
+    SWIFT = 'swift',
+    TYPESCRIPT = 'typescript',
+    VUE = 'vue',
+    ZIG = 'zig',
+}
+
+// Add different values to the Kind if using a different library.
+enum Kind {
+    MDI = 'mdi',
+    DEVICON = 'devicon',
+}
+
+interface FileInfo {
+    extension: FileExtension
+    // @TODO: render test files with test-indicator
+    isTest: boolean
+}
+
+// getFileInfo extracts the file extension from the full file name
+export const getFileInfo = (file: string): FileInfo => {
+    const extension = file.split('.').at(-1)
+    const isValidExtension = Object.values(FileExtension).includes(extension as FileExtension)
+
+    if (extension && isValidExtension) {
+        return {
+            extension: extension as FileExtension,
+            isTest: containsTest(file),
+        }
+    }
+
+    return {
+        extension: 'default' as FileExtension,
+        isTest: false,
+    }
+}
+
+interface IconSvelte {
+    info: IconInfo
+    // color ignored if 'kind' is 'devicon'
+    color: string
+}
+
+interface IconInfo {
+    value: string
+    kind: 'mdi' | 'devicon'
+}
+
+const getIconSvelte = (extension: FileExtension): IconSvelte => {
+    return {
+        info: getIconAttributes(extension),
+        color: getColor(extension),
+    }
+}
+
+const getIconAttributes = (extension: string): IconInfo => {
```

Comment:
This should be typed as `FileExtension` to get type checker support.

```suggestion
const getIconAttributes = (extension: FileExtension): IconInfo => {
```